x-pack/plugins/apm/server/lib/helpers/get_kuery_fields.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { KueryNode } from '@kbn/es-query';
+import { flatten } from 'lodash';
+
+export function getKueryFields(nodes: KueryNode[]) {
```

Comment:
I would love to have a helper from `es-query` for that job. 
Due to time constraint we can use the custom helper and create a follow up PR